### PR TITLE
#2920 Outbound returns detail view - skeleton

### LINKED ISSUE #2920: Returns -- supplier returns detail view
This is the page that loads after the items have been selected from the initial modal

(Similar to "Outbound shipment" UI)

Use mock API response

Broken out into further issues:
- [x] https://github.com/msupply-foundation/open-msupply/issues/3010
- [x] https://github.com/msupply-foundation/open-msupply/issues/3011
- [x] https://github.com/msupply-foundation/open-msupply/issues/3012
- [x] https://github.com/msupply-foundation/open-msupply/issues/3013
- [x] isDisabled hook
- [x] groupByItem functionality, including lines hook (lines also needed by StatusChangeButton)
- [x] enable delete mutation
- [x] Log tab

### PR DESCRIPTION
**Pull request by CarlosNZ**
Fixes #2920 (just the start really, won't close the issue)

# 👩🏻‍💻 What does this PR do? 
 

This has the barest bones of the Detail page. I think we'll do a separate PR for each component, as it's pretty slow finding all the pieces and creating the queries to make each one work (for me anyway).  This uses the single `invoiceByNumber` query, but it needs an additional query to get the lines.

Have left in but commented out the other elements that will need to appear (Toolbar, Sidebar, Footer, etc).

Created the route, but we might want to change it. It's currently: http://localhost:3003/distribution/outbound-return/1

Also, if anyone knows how to format the text for the breadcrumb header, please let me know -- it's currently just showing what's in the URL:
Screenshot 2024-02-15 at 4 09 17 PM

### CODE REVIEW

**Comment by lache-melvin** [comment]
> Also, if anyone knows how to format the text for the breadcrumb header, please let me know -- it's currently just showing what's in the URL:
> Screenshot 2024-02-15 at 4 09 17 PM

Add the route part to `app.json` as a translation string!
```json
"outbound-return": "Outbound Returns",
```

File: client/packages/invoices/src/InvoiceService.tsx [hunk 1]
```diff
@@ -31,6 +32,12 @@ const InvoiceService: FC = () => {
     .addPart(':invoiceNumber')
     .build();
 
+  // const outboundReturnsRoute = RouteBuilder.create(AppRoute.Returns).build();
```
Comment (lache-melvin) [inline client/packages/invoices/src/InvoiceService.tsx:35]: remove now?

**Review by lache-melvin — APPROVED** [review]
Looking good! Sounds like a good plan to get this much in and keep chipping away in smaller PRs :)

File: client/packages/invoices/src/Returns/api/hooks/index.ts [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import { Lines } from './line';
+import { Document } from './document';
 import { Utils } from './utils';
 
 export const useReturns = {
+  document: {
+    invoiceByNumber: Document.useOutboundReturn,
```
[the remark this answers is not in the document]
Comment (CarlosNZ) [reply client/packages/invoices/src/Returns/api/hooks/index.ts:7]: Yeah sure, have called it `outboundReturn` for now, we can maybe look at renaming them further if/when we simplify the api structure?

**Review by lache-melvin — APPROVED** [review]
Looks great thanks Carl! Let's get this in 😎

**Pull request merged**